src/v/cloud_topics/reconciler/reconciler.cc

```diff
@@ -473,7 +470,13 @@ reconciler::build_object(
               max_object_size);
             break;
         }
-        ctx.size_budget = max_object_size - current_size;
+        // It shouldn't happen, but beware underflow.
```

Comment:
This makes it seem defensive, but it can happen in the scenario described in the cover letter